src/state_lexcounter.erl

```diff
@@ -0,0 +1,243 @@
+    {ok, delta_state_lexcounter()}.
+delta_mutate(increment, Actor, {?TYPE, LexCounter}) ->
+    {Left, Right} = case orddict:find(Actor, LexCounter) of
+        {ok, Value} ->
+            Value;
+        error ->
+            {0, 0}
+    end,
+    Delta = orddict:store(Actor, {Left, Right + 1}, orddict:new()),
+    {ok, {?TYPE, {delta, Delta}}};
+
+delta_mutate(decrement, Actor, {?TYPE, LexCounter}) ->
+    {Left, Right} = case orddict:find(Actor, LexCounter) of
+        {ok, Value} ->
+            Value;
+        error ->
+            {0, 0}
+    end,
+    Delta = orddict:store(Actor, {Left + 1, Right - 1}, orddict:new()),
+    {ok, {?TYPE, {delta, Delta}}}.
+
+%% @doc Returns the value of the `state_lexcounter()'.
+%%      A `state_lexcounter()' is a dictionary where the values are
+%%      pairs. The value of the `state_lexcounter()' is the sum of
+%%      the second components of these pairs.
+-spec query(state_lexcounter()) -> non_neg_integer().
+query({?TYPE, LexCounter}) ->
+    lists:sum([ Right || {_Actor, {_Left, Right}} <- LexCounter ]).
+
+%% @doc Merge two `state_lexcounter()'.
+%%      The keys of the resulting `state_lexcounter()' are the union of the
+%%      keys of both `state_lexcounter()' passed as input.
+%%      If a key is only present on one of the `state_lexcounter()',
+%%      its correspondent value is preserved.
+%%      If a key is present in both `state_lexcounter()', the new value
+%%      will be the lexicographic join of both values.
+-spec merge(delta_or_state(), delta_or_state()) -> delta_or_state().
+merge({?TYPE, {delta, Delta1}}, {?TYPE, {delta, Delta2}}) ->
+    {?TYPE, DeltaGroup} = ?TYPE:merge({?TYPE, Delta1}, {?TYPE, Delta2}),
+    {?TYPE, {delta, DeltaGroup}};
+merge({?TYPE, {delta, Delta}}, {?TYPE, CRDT}) ->
+    merge({?TYPE, Delta}, {?TYPE, CRDT});
+merge({?TYPE, CRDT}, {?TYPE, {delta, Delta}}) ->
+    merge({?TYPE, Delta}, {?TYPE, CRDT});
+merge({?TYPE, LexCounter1}, {?TYPE, LexCounter2}) ->
+    LexCounter = orddict:merge(
+        fun(_, Value1, Value2) ->
+            lex_join(Value1, Value2)
+        end,
+        LexCounter1,
+        LexCounter2
+    ),
+    {?TYPE, LexCounter}.
+
+lex_join({Left1, Right1}, {Left2, _Right2}) when Left1 > Left2 ->
+    {Left1, Right1};
+lex_join({Left1, _Right1}, {Left2, Right2}) when Left2 > Left1 ->
+    {Left2, Right2};
+lex_join({Left1, Right1}, {Left2, Right2}) when Left1 == Left2 ->
+    {Left1, max(Right1, Right2)};
+lex_join({Left1, _Right1}, {Left2, _Right2}) ->
+    {max(Left1, Left2), 0}.
+
+%% @doc Equality for `state_lexcounter()'.
+-spec equal(state_lexcounter(), state_lexcounter()) -> boolean().
+equal({?TYPE, LexCounter1}, {?TYPE, LexCounter2}) ->
+    Fun = fun({Left1, Right1}, {Left2, Right2}) -> Left1 == Left2 andalso Right1 == Right2 end,
+    orddict_ext:equal(LexCounter1, LexCounter2, Fun).
+
+%% @doc Given two `state_lexcounter()', check if the second is and inflation
+%%      of the first.
+%% @todo
```

Comment:
@CBaquero 
Here I'm using the generic inflation check.
But what we discussed was `is_inflation((A, B), (A', B'))`
returns true if
- `A == A'` and `is_inflation(B, B')`
- `is_inflation(A, A')`

right?